src/gt4py/next/program_processors/runners/dace_fieldview/gtir_builtin_translators.py

```diff
@@ -194,55 +194,33 @@ def translate_as_field_op(
-    if isinstance(last_node, dace.nodes.Tasklet):
-        # the last transient node can be deleted
-        last_node_connector = state.in_edges(output_expr.node)[0].src_conn
-        state.remove_node(output_expr.node)
-    else:
-        last_node = output_expr.node
-        last_node_connector = None
-
-    # allocate local temporary storage for the result field
-    result_field = _create_temporary_field(sdfg, state, domain, node.type, output_desc)
+    taskgen = gtir_dataflow.LambdaToDataflow(sdfg, state, sdfg_builder, reduce_identity)
+    input_edges, output = taskgen.visit(stencil_expr, args=stencil_args)
+    output_desc = output.expr.node.desc(sdfg)
 
-    # assume tasklet with single output
-    output_subset = [dace_gtir_utils.get_map_variable(dim) for dim, _, _ in domain]
-    if isinstance(output_desc, dace.data.Array):
-        # additional local dimension for neighbors
+    domain_index = sbs.Indices([dace_gtir_utils.get_map_variable(dim) for dim, _, _ in domain])
+    if isinstance(node.type.dtype, itir_ts.ListType):
+        assert isinstance(output_desc, dace.data.Array)
         assert set(output_desc.offset) == {0}
-        output_subset.extend(f"0:{size}" for size in output_desc.shape)
+        # additional local dimension for neighbors
+        output_subset = sbs.Range.from_indices(domain_index) + sbs.Range.from_array(output_desc)
+    else:
+        assert isinstance(output_desc, dace.data.Scalar)
+        output_subset = sbs.Range.from_indices(domain_index)
 
     # create map range corresponding to the field operator domain
     map_ranges = {dace_gtir_utils.get_map_variable(dim): f"{lb}:{ub}" for dim, lb, ub in domain}
     me, mx = sdfg_builder.add_map("field_op", state, map_ranges)
 
-    if len(input_connections) == 0:
-        # dace requires an empty edge from map entry node to tasklet node, in case there no input memlets
-        state.add_nedge(me, last_node, dace.Memlet())
-    else:
-        for data_node, data_subset, lambda_node, lambda_connector in input_connections:
-            memlet = dace.Memlet(data=data_node.data, subset=data_subset)
-            state.add_memlet_path(
-                data_node,
-                me,
-                lambda_node,
-                dst_conn=lambda_connector,
-                memlet=memlet,
-            )
-    state.add_memlet_path(
-        last_node,
-        mx,
-        result_field.data_node,
-        src_conn=last_node_connector,
-        memlet=dace.Memlet(data=result_field.data_node.data, subset=",".join(output_subset)),
-    )
+    # allocate local temporary storage for the result field
+    result_field = _create_temporary_field(sdfg, state, domain, node.type, output_desc)
+
+    # here we setup the edges from the map entry node
+    for edge in input_edges:
+        edge.connect(me)
+
+    # and here the edge writing the result data through the map exit node
+    output.connect(mx, result_field.data_node, output_subset)
```

Comment:
I would add an overload to the connect function such that you could write `output.connect(mx, result_field, output_subset)` i.e. that the function figures this out on its own.
It is a bit more high level.

Put this is probably the C++ programmer speaking that misses true function overloading.